### ISSUE TITLE
book summary sharing social network website

### ISSUE THREAD

**Issue by MarshalOfficial** [open-source-ideas/open-source-ideas#266]
## Project description

A social network website, that gives members the ability to share the book summary of their learning journey, it has followed, star features also, in my personal case I am saving my learning journey on the Telegram messenger's private channels, it might be interesting that we make a place for developers to share their learning journey and of course it gives a push motivation on ourselves when we have some audience that waiting for every topic of the books in summary.
it will be special for computer science maybe.

## Relevant Technology

It will be a fully dynamic website, that might use any full-featured frameworks for example Net-Core and MongoDB in the backend and React js for the front-end.

- [ ] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [* ] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [* ] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [ ] Little work - A couple of days
- [ *] Medium work - A week or two
- [* ] Much work - The project will take more than a couple of weeks and serious planning is required

### Categories

- [ ] Mobile app
- [ ] IoT
- [* ] Web app
- [ *] Frontend/UI
- [ ] AI/ML
- [ *] APIs/Backend
- [ ] Voice Assistant
- [ ] Developer Tooling
- [ ] Extension/Plugin/Add-On
- [ ] Design/UX
- [ ] AR/VR
- [ ] Bots
- [ ] Security
- [ ] Blockchain
- [ ] Futuristic Tech/Something Unique

**Comment by IlanLuci** [comment]
Really love the idea! I don't have a ton of experience but would love to help out.

**Comment by KOLANICH** [comment]
I completely don't understand what the idea is because I completely don't understand what you mean under `the book summary of their learning journey`.

**Comment by KaKi87** [comment]
I'm not sure either.

The title makes me think about IMDB but for books, but I'm lost when reading the description 😅

**Comment by FredrikAugust** [comment]
Is it something like Goodreads?

**Comment by IlanLuci** [comment]
I read it as like a platform to document your progress learning something for example programming. Like how some people will write a biography or a book of some sort on how they progressed. Something like that but as a web interface where you can document and share your learning progress.

**Comment by MarshalOfficial** [comment]
Let me give you an example:
Imagine I start reading "The Pragmatic Programmer" book. While reading it has tons of examples and of course, some of them are the author's excess. because if you want to teach somebody the goal of each topic, you can summarize it in at most 10 sentences or a paragraph. 
so it needs a place where everyone can summarize their studies and be used by others.

**Comment by KOLANICH** [comment]
Already discussed, #53, https://github.com/metacademy

**Comment by gaborszita** [comment]
I'd be happy to help out. I'm fluent in Java, C++, HTML, and CSS and have some experience in MySQL, PHP, and Javascript. So when are we going to plan out this project? I think we should create a repository for this project and start planning the structure of it.

**Comment by MarshalOfficial** [comment]
> I'd be happy to help out. I'm fluent in Java, C++, HTML, and CSS and have some experience in MySQL, PHP, and Javascript. So when are we going to plan out this project? I think we should create a repository for this project and start planning the structure of it.

great @gaborszita 
I think, first, we need to document the features and facilities,
which platform do you prefer for this purpose?
maybe Trello or etc.

**Comment by gaborszita** [comment]
Hi @MarshalOfficial ,
I really don't care about the platform since I've never actually used one. Anything is good to me, so we could use Trello as you said. I'm happy to adapt to your needs. Anything that fits our uses is fine.

Correct me if I'm wrong, so this project's goal will be to give people a place where they can describe what they learnt from a book? Your description about this project is just not fully clear to me.

**Comment by MarshalOfficial** [comment]
> Hi @MarshalOfficial ,
> I really don't care about the platform since I've never actually used one. Anything is good to me, so we could use Trello as you said. I'm happy to adapt to your needs. Anything that fits our uses is fine.
> 
> Correct me if I'm wrong, so this project's goal will be to give people a place where they can describe what they learnt from a book? Your description about this project is just not fully clear to me.

Exactly, yes. the project goal is to let people share a summary of a book that they read in this.
great. so we can start in Trello documenting.
for the tech stack, what are your preferred backend framework and DB engine, and what for front-end?

**Comment by gaborszita** [comment]
Hi,

My preferred backend framework is PHP, database is MySQL, and front-end
HTML+CSS+JS.

**Comment by gaborszita** [comment]
My Trello account is on gaborszita88@gmail.com email, so use that email address if you'd like to add me to your Trello workspace.

**Comment by MarshalOfficial** [comment]
> My Trello account is on [gaborszita88@gmail.com](mailto:gaborszita88@gmail.com) email, so use that email address if you'd like to add me to your Trello workspace.

Sure, thanks

**Comment by RichardDev** [comment]
Hi do have you started something about this project?

**Comment by gaborszita** [comment]
Yes, I did. I started the project [gaborszita/snappy-book-review](https://github.com/gaborszita/snappy-book-review) .

I wrote the "prototype" of this project is using the LAMP stack. You can check it out on [snappybookreview.gaborszita.net](https://snappybokreview.gaborszita.net/) . 
I'm rewriting the project to use the TypeScript+ExpressJS+MongoDB stack (hey, it's 2022, LAMP is an old technology), which can be found on the `new-ts-node-app` branch of the repo. I worked on it in December, but don't currently have time for it due to school. I'll probably get back to it in June and finish it in the summer.

**Comment by MarshalOfficial** [comment]
> Yes, I did. I started the project [gaborszita/snappy-book-review](https://github.com/gaborszita/snappy-book-review) .
> 
> I wrote the "prototype" of this project is using the LAMP stack. You can check it out on [snappybookreview.gaborszita.net](https://snappybookreview.gaborszita.net/) . I'm rewriting the project to use the TypeScript+ExpressJS+MongoDB stack (hey, it's 2022, LAMP is an old technology), which can be found on the `new-ts-node-app` branch of the repo. I worked on it in December, but don't currently have time for it due to school. I'll probably get back to it in June and finish it in the summer.

great job, very nice prototype. but it must be used for book summary sharing and not just review.
for the tech stack, it is very good to use newer technologies than PHP

**Comment by gaborszita** [comment]
> > Yes, I did. I started the project [gaborszita/snappy-book-review](https://github.com/gaborszita/snappy-book-review) .
> > I wrote the "prototype" of this project is using the LAMP stack. You can check it out on [snappybookreview.gaborszita.net](https://snappybookreview.gaborszita.net/) . I'm rewriting the project to use the TypeScript+ExpressJS+MongoDB stack (hey, it's 2022, LAMP is an old technology), which can be found on the `new-ts-node-app` branch of the repo. I worked on it in December, but don't currently have time for it due to school. I'll probably get back to it in June and finish it in the summer.
> 
> great job, very nice prototype. but it must be used for book summary sharing and not just review. for the tech stack, it is very good to use newer technologies than PHP

Hi,
yes, that's why I called it a prototype. It was mainly just for designing the front-end, meaning how the website would look. It doesn't even have vital functions like deleting reviews a user created.
I forgot to mention that I'm using bootstrap (I suck at CSS), but I'm using it in a way that if we want to use our own style on it someday, we can easily modify it to use our style.

**Comment by gaborszita** [comment]
Update:
It's summer break, I'm working heavily on Snappy Book Review. Basic account control is working, I'm currently working to expand it to full account control. When it's done, we can use SBR as the template for this project and create a fork, so book search and account control will already be done.

@MarshalOfficial how do you plan that people can summarize a topic of a book? Will we list the book contents and people can add a summary to each topic?

**Comment by TAKANOME-DEV** [comment]
@gaborszita why not switch to monorepo and use Frontend frameworks for the client side??

**Comment by MarshalOfficial** [comment]
> Update: It's summer break, I'm working heavily on Snappy Book Review. Basic account control is working, I'm currently working to expand it to full account control. When it's done, we can use SBR as the template for this project and create a fork, so book search and account control will already be done.
> 
> @MarshalOfficial how do you plan that people can summarize a topic of a book? Will we list the book contents and people can add a summary to each topic?

Good to hear that you are working on the project. I think because of the copywriter issue we can not put book content and just people's summary is enough

**Comment by gaborszita** [comment]
Okay, in that case, we can simply just put this summary-sharing feature in Snappy Book Review and rename the project Snappy Book Portal (since it won't be a review-only site). Hence, we won't have to create another fork and it will be easier to maintain.
@TAKANOME-DEV what do you mean by using front-end what do you mean by using front-end frameworks for the client side? I didn't use any frameworks, like React, because it's a multi-page site didn't need them. I could've used JQuery, but vanilla JS is just better in 2022.

**Comment by gaborszita** [comment]
Update: People can now post summaries on books:

![image](https://user-images.githubusercontent.com/30361931/183304604-4cc93ebc-0569-44ac-a11f-a6a2ac94a014.png)

**Comment by jeel279** [comment]
I'm interested if it is still available. I'm have prior knowledge of C/C++,Java,Kotlin, NodeJS, Typescript. Also I know javascript frontend/serverless architecture like NextJS. Please let me know if it is still available.
